src/main.rs

```diff
@@ -283,7 +282,8 @@ fn start_vmm(cmd_arguments: ArgMatches) {
     } else {
         SeccompLevel::Advanced
     };
-    let hypervisor = Hypervisor::new().unwrap();
+    #[cfg(feature = "kvm")]
+    let hypervisor = hypervisor::kvm::KvmHypervisor::new().unwrap();
```

Comment:
Thinking how the vmm crate could eventually become a standalone rust-vmm crate, I'd like to make it as easy as possible to use it. The above code will eventually become a repeating pattern for any VMM using the hypervisor crate:

```Rust
#[cfg(feature = "hypervisor-foo")]
let hv = hypervisor::kvm::FooHypervisor::new().unwrap();
#[cfg(feature = "hypervisor-bar")]
let hv = hypervisor::hyperv::BarHypervisor::new().unwrap();

hv: Arc<dyn hypervisor::Hypervisor> = Arc::new(hv);
```

Ideally, a VMM could pass its selected hypervisor to the hypervisor crate as a feature and be done with it.

So having a hypervisor constructor in `hypervisor/src/lib.rs` like e.g.:

```Rust
pub fn new() -> hypervisor::Result<dyn hypervisor::Hypervisor> {
        #[cfg(feature = "kvm")]
        kvm::KvmHypervisor::new()
}
```

Or a more exclusive option:

```Rust
#[cfg(feature = "kvm")]
pub use kvm::KvmHypervisor as Hypervisor;
```

would be preferable to me.
I know it contradicts with https://github.com/cloud-hypervisor/cloud-hypervisor/pull/1433#discussion_r450779923 and I'm late to the game, but I think this would:
* Make the hypervisor crate easier to use
* Make the VMM code truly hypervisor agnostic

@sboeuf WDYT?

Reply:
@sameo  I prefer this one: 
#[cfg(feature = "kvm")]
pub use kvm::KvmHypervisor as Hypervisor;

Actually in our internal integration of hyperv with CH I am using this implementation.

Reply:
I'm happy to use any approach that is agreed upon.

If we do `pub use kvm::KvmHypervisor as Hypervisor` we will need to come up with a new name for the Hyperivosr trait.

Reply:
@liuw @russell-islam I'd prefer what I think is a more flexible option:

```Rust
pub fn new() -> hypervisor::Result<Arc<dyn hypervisor::Hypervisor>> {
    #[cfg(feature = "kvm")]
    let hypervisor: Arc<dyn hypervisor::Hypervisor> = Arc::new(kvm::KvmHypervisor::new()?);

    Ok(hypervisor)
}
```

Reply:
Works for me. I already wrote the exact function yesterday. :-)

I will wait for @sboeuf to voice his opinion.

Reply:
Sorry for the late answer here. I had a discussion offline with @sameo about this. It looks fine to me since we don't need to do anything KVM or Hyper-V specific from the VMM. As long as everything can be abstracted behind the Hypervisor trait, then I'm okay having the hypervisor crate taking care of the Hypervisor creation, depending on the rust feature `kvm` or `hyperv`.

Reply:
Thanks for the reply. I will rework my branch to address comments made so far.